lib/ex_doc/retriever.ex

```diff
@@ -58,22 +58,21 @@ defmodule ExDoc.Retriever do
   # with --docs flag), we raise an exception.
   defp get_module(module, config) do
     if docs_chunk = docs_chunk(module) do
-      generate_node(module, docs_chunk, config)
+      module_data = get_module_data(module, docs_chunk)
+
+      if not module_data.skip do
+        [generate_node(module, module_data, config)]
+      else
+        []
+      end
     else
       []
     end
   end
 
-  defp nesting_info(title, prefixes) do
-    prefixes
-    |> Enum.find(&String.starts_with?(title, &1 <> "."))
-    |> case do
-      nil -> {nil, nil}
-      prefix -> {String.trim_leading(title, prefix <> "."), prefix}
-    end
-  end
-
-  # Special case required for Elixir
+  # TODO: see if we can do something about it after all?
+  # Special case required for Elixir - we don't know it's Elixir until we read
+  # the chunk so we can't have ExDoc.Language.Elixir specific behaviour.
   defp docs_chunk(:elixir_bootstrap), do: false
```

Comment:
I think it is fine to do nothing about this as I think both of them are optimizations?